stickermanager.py

```diff
@@ -0,0 +1,354 @@
+from telethon import events, utils, TelegramClient
+
+POLL_TEMPLATE = (
+    '<a href="tg://user?id={sender_id}">{sender_name}</a> has suggested this '
+    'sticker be added to <a href="{pack_link}">the group\'s sticker pack</a> '
+    'with the emoji {emoji}.\n\n'
+    '<strong>Current result: {score}</strong>\n'
+    'For: {yes}\n'
+    'Against: {no}'
+)
+
+POLL_FINISHED_TEMPLATE = (
+    '<strong>This sticker poll finished with a final score of {score}. {result}</strong>\n\n'
+    '<a href="tg://user?id={sender_id}">{sender_name}</a> had suggested this '
+    'sticker be added to the group\'s sticker pack with the emoji {emoji}.\n\n'
+    'For: {yes}\n'
+    'Against: {no}'
+)
+
+VOTE_TEMPLATE = '<a href="tg://user?id={uid}">{displayname}</a> ({weight})'
+
+RESULT_ADDED = 'The sticker has been added to <a href="{pack_link}">the pack</a>.'
+RESULT_REJECTED = 'The sticker was rejected from <a href="{pack_link}">the pack</a>.'
+
+UP = '\U0001f53c'
+DOWN = '\U0001f53d'
+UP_DAT = b'addsticker/+'
+DOWN_DAT = b'addsticker/-'
+
+POLL_TIMEOUT = ADD_COOLDOWN = 24 * 60 * 60
+
+BASE_DIR = Path(__file__).resolve().parent
+CONFIG_FILE = BASE_DIR.joinpath("stickermanager.tsv")
+CACHE_FILE = BASE_DIR.joinpath("stickermanager.json")
+DATA_FILE_FORMAT: str = "stickermanager.{ts}.dat"
+
+VoteData = NamedTuple('VoteData', weight=int, displayname=str)
+Number = Union[int, float]
+Scores = NamedTuple('Scores', sum=Number, yes=Number, no=Number, yes_count=int, no_count=int)
+
+current_vote: Optional[dict] = None
+current_vote_lock: asyncio.Lock = asyncio.Lock()
+current_vote_status: asyncio.Event = asyncio.Event()
+last_accepted: int = 0
+sticker_pack = None
+
+with open(CONFIG_FILE) as f:
+    WEIGHTS = {
+        int(uid) if uid.strip().isnumeric() else uid:
+            (0 if weight == '-' else float(weight), None if name == '-' else name)
+        for uid, weight, name
+        in (line.strip().split('\t', 2)
+            for line in f
+            if line and not line.startswith('#') and not line.isspace())
+    }
+
+    VOTES_REQUIRED = WEIGHTS.pop('votes required')[0]
+    DEFAULT_WEIGHT = WEIGHTS.pop('default weight')[0]
+    STICKER_PACK_TITLE = WEIGHTS.pop('sticker title')[1]
+    STICKER_PACK_SHORT_NAME = WEIGHTS.pop('sticker short')[1]
+    ADMIN_USER_ID = int(WEIGHTS.pop('sticker owner')[1])
+    ALLOWED_CHATS = [int(gid)
+                     for gid in WEIGHTS.pop('allowed chats')[1].split(',')
+                     if not gid.isspace()]
+
+
+def load_cache() -> None:
+    global sticker_pack, current_vote, last_accepted
+    with open(CACHE_FILE) as file:
+        data = json.load(file)
+        sp_data = data['sticker_pack']
+        if sp_data:
+            sticker_pack = InputStickerSetID(id=sp_data['id'], access_hash=sp_data['access_hash'])
+        cv_data = data['current_vote']
+        if cv_data:
+            current_vote = cv_data
+            current_vote["votes"] = {int(uid): VoteData(*data)
```

Comment:
How do you judge whether to use double or single quotes? Given you made use of single quotes everywhere else (even when a string already used them, requiring escaping), shouldn't this be:
```suggestion
            current_vote['votes'] = {int(uid): VoteData(*data)
```

Reply:
I used double quotes everywhere and then did `s/"/'/g/` because you use single quotes